### DISCUSSION TITLE
Best way to traverse NGT tree

### DISCUSSION THREAD

**Discussion by IRobot1** [nartc/angular-three#144] [General]
What's the best way to traverse the tree of NGT objects from ngt-canvass down?  Is there any existing code that is a good starting point?

I was thinking of having a way to visualize this inside a scene and have lines that go from the NGT tree node visual to the three object.  This component could be added as a potential debug/helper aid.

This could also be done for the three scene tree objects too, but will require objects to have meaningful names to make it useful.

**Comment by nartc** [comment]
Does Angular Dev Tools work for you? (https://angular-three.netlify.app/docs/core/canvas)
<img width="977" alt="image" src="https://user-images.githubusercontent.com/25516557/183250211-049a52d4-e235-455a-ab1e-cc9fc49ea2e3.png">

**Reply by IRobot1** [reply]
Yes.  What I'd like to try is visualizing the equivalent of this as a NGT component in the scene.  I'm looking for help on how to traverse the NGT tree from within an NGT component.
